lnd_test.go

```diff
@@ -1048,8 +1048,17 @@ func testDisconnectingTargetPeer(net *lntest.NetworkHarness, t *harnessTest) {
 
 	// Disconnect Alice-peer from Bob-peer without getting error
 	// about existing channels.
-	if err := net.DisconnectNodes(ctxt, net.Alice, net.Bob); err != nil {
-		t.Fatalf("unable to disconnect Bob's peer from Alice's: err %v", err)
+	var predErr error
+	err = lntest.WaitPredicate(func() bool {
```

Comment:
Why's this necessary now?

Reply:
Because we now don't consider the channel closed (adding the `CloseSummary` to the db) before the commitment is on-chain, we run into a race with the above force close. Alternative is to modify the `DisconnectPeer` rpc to allow disconnecting from a peer where the commitment has been broadcasted, but not confirmed yet?